chore(ci): adjust labeler job for external contributors, in order not to rely on an octo-sts token

philipreinken opened the pull request:
My [previous PR](https://github.com/shopware/shopware/pull/10290) relied on generating an octo-sts token which can be used to query org and team membership; this does not work for accounts that don't have write permission on the shopware repo.

Instead, the job now checks the [repository permissions](https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user) to see, whether the triggering account has write access, which would apply to org members and invited collaborators.

Test run: https://github.com/shopware/shopware/pull/10449#discussion_r2142667370

philipreinken committed:
chore(ci): adjust labeler job for external contributors, in order not to rely on another action and an octo-sts token

Resolves: #9682

mitelg commented on .github/workflows/01-pr-issue-labeler.yml:
```diff
@@ -79,44 +79,43 @@ jobs:
+              return;
+            }
 
             const response = await github.rest.issues.addLabels({
```
who or what sets the label in the end with which permissions? because I am not sure, if externals are actually able to set any labels :thinking:

philipreinken replied:
The label will be set using the `GITHUB_TOKEN` (not tied to any user account), which has write access to pull-requests.

philipreinken replied:
We just talked about this in our daily - the token might not be tied to any user, but the permissions would still be limited by the triggering user account.

I'll try to find another way around this.

philipreinken replied:
I'm now using the `pull_request_target` event, which prevents code of forks to be checked out by default, while providing more permissions to the `GITHUB_TOKEN`.

philipreinken replied:
Alright, I've tested this implementation now - I've merged this branch to `trunk` in a fork of mine, creating a PR using [my regular account](https://github.com/philipreinken/shopware-test-fork/pull/4) as well as a [test account](https://github.com/philipreinken/shopware-test-fork/pull/3) which is no collaborator on the fork.

With my own account, the label is skipped: https://github.com/philipreinken/shopware-test-fork/actions/runs/15610932570/job/43971418913?pr=4#step:2:44
While it's set when using the test account without any privileges on the fork: https://github.com/philipreinken/shopware-test-fork/actions/runs/15610873551/job/43971216020?pr=3#step:2:44

mitelg replied:
nice!

philipreinken committed:
chore(ci): use pull_request_target event, so we're able to set labels

Refs: #9682

pweyck reviewed (dismissed):
we should have the `pull_request_target` jobs in a different workflow, to avoid running jobs by accident with high permissions

philipreinken committed:
chore(ci): create a separate workflow for the external-contribution label

This prevents us from accidentally running other jobs with more privileges than needed

Refs: #9682

philipreinken commented:
> we should have the `pull_request_target` jobs in a different workflow, to avoid running jobs by accident with high permissions

Good point, I've moved the job into a separate workflow now.
